api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1859,6 +1859,97 @@ type AzurePlatformSpec struct {
 	// +immutable
 	// +required
 	SecurityGroupID string `json:"securityGroupID,omitempty"`
+
+	// ManagedIdentities contains the client IDs related to the managed identities needed for HCP control plane
+	// and data plane components that authenticate with Azure's API.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	ManagedIdentities *AzureResourceManagedIdentities `json:"managedIdentities,omitempty"`
```

Comment:
Required structs should not be pointers
```suggestion
	ManagedIdentities AzureResourceManagedIdentities `json:"managedIdentities,omitempty"`
```